src/rgw/rgw_bucket.cc

```diff
@@ -1165,7 +1164,8 @@ int RGWBucket::check_object_index(RGWBucketAdminOpState& op_state,
   Formatter *formatter = flusher.get_formatter();
   formatter->open_object_section("objects");
   while (is_truncated) {
-    map<string, rgw_bucket_dir_entry> result;
+    RGWRados::ent_map_t result;
+    result.reserve(1000);
```

Comment:
Given the number of times "1000" appears in the code, I'm wondering if a `constexpr` would be in order....